qiskit/transpiler/passes/routing/algorithms/token_swapper.py

```diff
@@ -154,33 +151,33 @@ def swap(node0: _V, node1: _V) -> None:
             # Note that if there are only unhappy swaps involving this todo_node,
             # then an unhappy swap must be performed at some point.
             # So it is not useful to globally search for all happy swap chains first.
-            try:
-                cycle = nx.find_cycle(sub_digraph, source=todo_node)
+            cycle = rx.digraph_find_cycle(sub_digraph, source=todo_node)
```

Comment:
Is it necessary to prefix digraph function by `digraph_`?

Reply:
It's unfortunately a limitation with retworkx right now. We haven't figured out a pattern over there for a function to accept both  a `rx.PyGraph` or an `rx.PyDiGraph` since they're different types in the rust code and getting the python binding lib to accept a generic type with specific traits has proven difficult. So, for a lot of functions they only accept one or the other and will `TypeError` otherwise we've been prefixing them with `digraph_` or `directed_` (but not all, for example: https://retworkx.readthedocs.io/en/stable/stubs/retworkx.cycle_basis.html#retworkx.cycle_basis and https://retworkx.readthedocs.io/en/stable/stubs/retworkx.lexicographical_topological_sort.html). I'm hoping eventually we'll be able to find a a way to have a `rx.find_cycle()` that works with both an undirected and directed graph types and we can deprecate all the split functions.

Reply:
I see. Maybe you could dispatch the right Rust code by switching on the type in the Python code? In any case, this is out of scope here.

Reply:
Yeah, that would probably work fine. I had been avoiding that approach since all the function definitions exist in the rust side of the package and I was hoping we'd be able to come up with an approach to do it natively. But, doing it on the Python side will definitely be faster to get it out there. Could you open an issue about this at https://github.com/Qiskit/retworkx/issues so we can track this? Then we probably can get this in for the next release.

Reply:
I've opened an issue.